src/test/java/org/folio/moduserstest/ProxyRelationshipsIT.java

```diff
@@ -0,0 +1,278 @@
+package org.folio.moduserstest;
+
+import static java.net.HttpURLConnection.HTTP_CREATED;
+import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
+import static java.net.HttpURLConnection.HTTP_NO_CONTENT;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.net.URI;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.folio.postgres.testing.PostgresTesterContainer;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.folio.support.ProxyRelationship;
+import org.folio.support.ValidationErrors;
+import org.folio.support.VertxModule;
+import org.folio.support.http.OkapiHeaders;
+import org.folio.support.http.ProxiesClient;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.vertx.core.Vertx;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import lombok.SneakyThrows;
+
+/**
+ * At the moment, proxy relationships do not require the user to exist
+ * this means that these tests do not currently create users for any of the
+ * users involved.
+ * This is NOT how the system is intended to be used
+ */
+@ExtendWith(VertxExtension.class)
+@Timeout(value = 20, unit = SECONDS)
+class ProxyRelationshipsIT {
+  private static ProxiesClient proxiesClient;
+
+  @BeforeAll
+  @SneakyThrows
+  public static void beforeAll(Vertx vertx, VertxTestContext context) {
+    PostgresClient.setPostgresTester(new PostgresTesterContainer());
+
+    int port = NetworkUtils.nextFreePort();
+
+    final var headers = new OkapiHeaders("http://localhost:" + port,
+      "diku", "diku");
+
+    proxiesClient = new ProxiesClient(new URI("http://localhost:" + port), headers);
+
+    final var module = new VertxModule(vertx);
+
+    module.deployModule(port)
+      .onComplete(context.succeeding(res -> module.enableModule(headers,
+          false, false)
+        .onComplete(context.succeedingThenComplete())));
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    proxiesClient.deleteAllProxies();
+  }
+
+  @Test
+  void userMayNominateAProxy() {
+    final var userId = generateId();
+    final var proxyUserId = generateId();
+
+    final var relationship = proxiesClient.createProxyRelationship(
+      ProxyRelationship.builder()
+        .userId(userId)
+        .proxyUserId(proxyUserId)
+        .build());
+
+    assertThat(relationship.getId(), is(notNullValue()));
+    assertThat(relationship.getUserId(), is(userId));
+    assertThat(relationship.getProxyUserId(), is(proxyUserId));
```

Comment:
Matching the rest-assured way:
```
    proxiesClient.createProxyRelationship(
       ProxyRelationship.builder()
         .userId(userId)
         .proxyUserId(proxyUserId)
        .build(),
      "id", is(notNullValue()),
      "userId", is(userId),
      "proxyUserId", is(proxyUserId));
```

It requires this `.body` call:
```
  public ProxyRelationship createProxyRelationship(
      @NonNull ProxyRelationship proxyRelationship,
      String path, Matcher<?> matcher, Object... pathMatcherPairs) {

     return attemptToCreateProxyRelationship(proxyRelationship)
        .statusCode(HTTP_CREATED)
        .body(path, matcher, pathMatcherPairs)
        .extract().as(ProxyRelationship.class);
  }
```

Reply:
I agree that RestAssured provides those capabilities. I disagree that it is _the_ rest assured way.

What is better about passing matchers into the client than expressing them separately?

I personally find it easier to understand the important aspects of the tests much of the mechanics of the API are abstracted away yet the pertinent checks are in the tests themselves.

I find it easier to read separate assertions and they tend to provide better feedback on failure, in that I know which specific assertion has failed.

There is a balance to be found in the use of RestAssured. A part of me would prefer to deserialise to a response as quickly as possible and do everything else separately. I decided that using RestAssured's status checking and body deseralization was a reasonable compromise.

This model also allows for model in the tests to adapt separately to the raw API's (partly to hide some unfortunate design decisions) and it supports refactoring tools better than using strings to refer to fields.